### PR TITLE
Initialize the Laravel Project & Setup initial CI pipeline

### LINKED ISSUE #6: Setup CI Environment

### PR DESCRIPTION
**Pull request by alexstojda**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

This PR initializes the Laravel project & has a basic CI pipeline. 

### Test plan (required)
n/a project initialization

### Closing issues
- Closes #6 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [ ] Static Analysis Rules Validated (not yet implemented)
- [x] All commits follow naming conventions

### CODE REVIEW

**Review by Alex-Hayeur — COMMENTED** [review]
How many of these files were generated by Laravel/CI and will they need to be reviewed as well?

**Comment by alexstojda** [comment]
> How many of these files were generated by Laravel/CI and will they need to be reviewed as well?

@Alex-Hayeur Basically all the code generated through the initialization is contained in a0359e2b1422283482741f08bb819347e71f53eb . The other commits contain the work done by me.

**Review by Alex-Hayeur — DISMISSED** [review]
The setup looks good with respect to Laravel's standard practices. One comment but it shouldn't have an impact so I can leave an approval.

Is it standard to keep the environment variable template as-is? I don't believe we'll be using all of the fields (such as AWS) but I'll double-check what the stakeholders will be using to be sure.

**Comment by alexstojda** [comment]
> The setup looks good with respect to Laravel's standard practices. One comment but it shouldn't have an impact so I can leave an approval.
> 
> Is it standard to keep the environment variable template as-is? I don't believe we'll be using all of the fields (such as AWS) but I'll double-check what the stakeholders will be using to be sure.

Since we are still in the early stages of the project I decided it would be best to leave them in as a comment for reference :)

**Comment by WindOfRussia** [comment]
> The setup looks good with respect to Laravel's standard practices. One comment but it shouldn't have an impact so I can leave an approval.
> 
> Is it standard to keep the environment variable template as-is? I don't believe we'll be using all of the fields (such as AWS) but I'll double-check what the stakeholders will be using to be sure.

Typical Laravel setups leave all the template env variables as is and just add new/custom ones on top of the original.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# PHP CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-php/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # Specify the version you desire here
+      - image: circleci/php:7.3-node-browsers
```
Comment (WindOfRussia) [inline .circleci/config.yml:10]: Any chance we can bump to 7.4 ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            - ./vendor
+
+      - run:
+          name: "Generate App key"
+          command: php artisan key:generate
+
+      # run tests with phpunit
+      - run:
+          name: "Run Unit Tests"
+          command: ./vendor/bin/phpunit tests
```
Comment (WindOfRussia) [inline .circleci/config.yml:52]: laravel 7+ has added a new artisan test command so no need to invoke phpunit directly.

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Update the VARIANT arg in docker-compose.yml to pick a PHP version: 7, 7.4, 7.3
+ARG VARIANT=7
```
Comment (WindOfRussia) [inline .devcontainer/Dockerfile:2]: Same as before, can we bump to 7.4?

File: public/.htaccess [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<IfModule mod_rewrite.c>
```
Comment (WindOfRussia) [inline public/.htaccess:1]: Not related to this file really but are we running on apache or nginx ?

**Review by WindOfRussia — APPROVED** [review]
Looks good for approval. Just leaving a note that we should bump php to 7.4 in order to support some of the standard dev packages like php-insights for linting purposes that will be added shortly.

**Review by danseremet — APPROVED** [review]
Clean initial project :+1: 
Very good for support of OS-independent development :100: :tada: 

One crucial missing thing is the important README.md file which includes an explanation on how to successfully get this project going on our local machines and details all (commonly used) commands necessary for starting the web server. (All of this could have its own relative link to a CONTRIBUTING.md file)

Things to mention
- docker (how to go into a bash terminal into your container for those unfamiliar with it e.g. `docker exec -it container-id bash`
- docker-compose (maybe?)
- information about running the web server (or if it's already run by default when the dev container starts, to specify it is)
- testing
- integration testing
- etc.. (usage of how development tools are used)

Also to add the other necessary tools to php artisan under the dev container e.g. telescope :eyes: 

**Note**: All of these can go into a new issue so I'll approve this PR.

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (danseremet) [reply .devcontainer/Dockerfile:2]: Should we follow a minor version instead of major?

File: public/.htaccess (hunk 4 above)
Comment (danseremet) [reply public/.htaccess:1]: image
Answer ^ :eyes:

Comment (danseremet) [reply public/.htaccess:1]: Should be nginx right @alexstojda ?

File: .circleci/config.yml (hunk 1 above)
Comment (alexstojda) [reply .circleci/config.yml:10]: Will bump to 7.4 but we will need to confirm its use with the stakeholders

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (alexstojda) [reply .devcontainer/Dockerfile:2]: Will bump to 7.4 but we will need to confirm its use with the stakeholders

File: public/.htaccess (hunk 4 above)
Comment (alexstojda) [reply public/.htaccess:1]: Will switch to nginx

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (alexstojda) [reply .devcontainer/Dockerfile:2]: #50

File: public/.htaccess (hunk 4 above)
Comment (alexstojda) [reply public/.htaccess:1]: Actually will open separate ticket for that, needs a bit of extra work to setup in the dev container #5

**Comment by alexstojda** [comment]
> Clean initial project 👍
> Very good for support of OS-independent development 💯 🎉
> 
> One crucial missing thing is the important README.md file which includes an explanation on how to successfully get this project going on our local machines and details all (commonly used) commands necessary for starting the web server. (All of this could have its own relative link to a CONTRIBUTING.md file)
> 
> Things to mention
> 
> * docker (how to go into a bash terminal into your container for those unfamiliar with it e.g. `docker exec -it container-id bash`
> * docker-compose (maybe?)
> * information about running the web server (or if it's already run by default when the dev container starts, to specify it is)
> * testing
> * integration testing
> * etc.. (usage of how development tools are used)
> 
> Also to add the other necessary tools to php artisan under the dev container e.g. telescope 👀
> 
> **Note**: All of these can go into a new issue so I'll approve this PR.

Good points @danseremet could you open a Task for that?

**Review by danseremet — APPROVED** [review]
:+1: https://github.com/CSU-Booking-Platform/application/issues/51

**Review by Alex-Hayeur — APPROVED** [review]
Confirming approval, I thought I already approved it.

**Pull request merged**